odoo/addons/base/models/res_users.py

```diff
@@ -1381,6 +1416,29 @@ def fields_get(self, allfields=None, attributes=None):
                     }
         return res
 
+class CheckIdentity(models.TransientModel):
+    _name = 'res.users.identitycheck'
+    _description = """ Wizard used to re-check the user's credentials (password)
+
+Might be useful before the more security-sensitive operations, users might be
+leaving their computer unlocked & unattended. Re-checking credentials mitigates
+some of the risk of a third party using such an unattended device to manipulate
+the account.
+"""
+
+    request = fields.Char(readonly=True, groups='.')
+    password = fields.Char()
+
+    def run_check(self):
+        if not request:
+            raise UserError(_("This method can only be accessed over HTTP"))
+        self.create_uid._check_credentials(self.password, {'interactive': True})
+        self.password = False
```

Comment:
shouldn't we flush here ?

Reply:
No? It should automatically be flushed at the first read or when the query ends, the point is just to "nuke" the password from this transient as the transient has been "consumed" by the check.